src/sympc/tensor/replicatedshare_tensor.py

```diff
@@ -123,14 +124,12 @@ def get_shares(self) -> List[torch.Tensor]:
     def sanity_checks(
         x: "ReplicatedSharedTensor",
         y: Union[int, float, torch.Tensor, "ReplicatedSharedTensor"],
-        op_str: str,
-    ) -> "ReplicatedSharedTensor":
+    ) -> Tuple["ReplicatedSharedTensor", Tuple[Session, int, Config, int, int]]:
```

Comment:
MAYBE: we could have another argument , something like boolean value named `parties_check` or `falcon_check` for ops like `mul`, `matmul` , if set to true, it will check ,if `number of parties !=3` , if session_uuid id not set, we could have `len(shares)!=2`, for sanity checks. The boolean value will default value `False`.

Reply:
Sanity check is native to RSTensor so would be good to keep protocol related information separate or wherever it comes into play (mul for 3 parties)